src/extension.ts

```diff
@@ -63,12 +82,29 @@ export function activate(context: ExtensionContext) {
     });
 
     workspace.onDidSaveTextDocument(doc => {
-        if (ameba && ameba.config.onSave) ameba.execute(doc);
+        if (ameba && ameba.config.onSave && doc.languageId === 'crystal' && !doc.isUntitled && doc.uri.scheme === 'file') {
```

Comment:
this could be moved a separate function:

```js
function isValidCrystalDocument(doc: TextDocument): boolean {
    return doc.languageId === 'crystal' && !doc.isUntitled && doc.uri.scheme === 'file';
}
```